src/odemis/driver/test/technolution_test.py

```diff
@@ -0,0 +1,1460 @@
+        if 'frequency' not in clockFrequencyData:
+            raise IOError("Could not obtain clock frequency, received data does not hold the proper key")
+        clock_freq = clockFrequencyData['frequency']
+
+        self.assertIsInstance(clock_freq, int)
+
+        self.assertEqual(
+                self.EBeamScanner.clockPeriod.value,
+                1 / clock_freq)
+
+    def test_resolutionVA(self):
+        min_res = self.EBeamScanner.resolution.range[0][0]
+        max_res = self.EBeamScanner.resolution.range[1][0]
+
+        # Check if small resolution values are allowed
+        self.EBeamScanner.resolution.value = (min_res + 5, min_res + 5)
+        self.assertEqual(self.EBeamScanner.resolution.value, (min_res + 5, min_res + 5))
+
+        # Check if big resolutions values are allowed
+        self.EBeamScanner.resolution.value = (max_res - 200, max_res - 200)
+        self.assertEqual(self.EBeamScanner.resolution.value, (max_res - 200, max_res - 200))
+
+        # Check if VA refuses to set limits outside allowed range
+        with self.assertRaises(IndexError):
+            self.EBeamScanner.resolution.value = (max_res + 10, max_res + 10)
+
+        self.assertEqual(self.EBeamScanner.resolution.value, (max_res - 200, max_res - 200))
+
+        with self.assertRaises(IndexError):
+            self.EBeamScanner.resolution.value = (min_res - 1, min_res - 1)
+        self.assertEqual(self.EBeamScanner.resolution.value, (max_res - 200, max_res - 200))
+
+        # Check if it is allowed to have non-square resolutions
+        self.EBeamScanner.resolution.value = (6000, 6500)
+        self.assertEqual(self.EBeamScanner.resolution.value, (6000, 6500))
+
+    def test_dwellTimeVA(self):
+        min_dwellTime = self.EBeamScanner.dwellTime.range[0]
+        max_dwellTime = self.EBeamScanner.dwellTime.range[1]
+
+        self.EBeamScanner.dwellTime.value = 0.9 * max_dwellTime
+        self.assertEqual(self.EBeamScanner.dwellTime.value, 0.9 * max_dwellTime)
+
+        self.EBeamScanner.dwellTime.value = min_dwellTime
+        self.assertEqual(self.EBeamScanner.dwellTime.value, min_dwellTime)
+
+        # Check if VA refuses to set limits outside allowed range
+        with self.assertRaises(IndexError):
+            self.EBeamScanner.dwellTime.value = 1.2 * max_dwellTime
+        self.assertEqual(self.EBeamScanner.dwellTime.value, min_dwellTime)
+
+        with self.assertRaises(IndexError):
+            self.EBeamScanner.dwellTime.value = 0.5 * min_dwellTime
+        self.assertEqual(self.EBeamScanner.dwellTime.value, min_dwellTime)
+
+    def test_getTicksDwellTime(self):
+        dwellTime = 0.9 * self.EBeamScanner.dwellTime.range[1]
+        self.EBeamScanner.dwellTime.value = dwellTime
+        self.assertIsInstance(self.EBeamScanner.getTicksDwellTime(), int)
+        self.assertEqual(self.EBeamScanner.getTicksDwellTime(), int(dwellTime / self.EBeamScanner.clockPeriod.value))
```

Comment:
Both are actually of type int and the method self.EBeamScanner.getTicksDwellTime() also rounds by using ints. So using ints makes the most sense. We could also use math.floor but since ticks are only in positive numbers ints is much more logical.

Reply:
ok